dvc/stage/__init__.py

```diff
@@ -791,12 +530,10 @@ def get_used_cache(self, *args, **kwargs):
 
 
 class PipelineStage(Stage):
-    def __init__(self, *args, name=None, meta=None, **kwargs):
+    def __init__(self, *args, name=None, **kwargs):
         super().__init__(*args, **kwargs)
         self.name = name
         self.cmd_changed = False
-        # This is how the Stage will discover any discrepancies
-        self.meta = meta or {}
```

Comment:
Note that this `meta` was never used before. The idea during the implementation was to provide metadata to the stage
on the discrepancies between lockfile and pipeline file (eg: someone added an out in pipeline file manually). But, since
we are moving away to having less warnings and messages, this _future-proofing_ is not needed.

Reply:
@skshetry So `meta` is no longer supported in `dvc.yaml`?

Reply:
No, `meta` is still supported (this is about metadata regarding discrepancies that was supposed to be tracked between lockfile and dvc.yaml file.

Eg: when someone manually added 'foo' and removed 'bar' from dvc.yaml, but didn't change lockfile, we could provide message to the user.
```python
meta = {'added': ['foo'], 'removed': ['bar']}
# 'bar' was removed on stage: 'generate-foo'
# lockfile needs adjustments
```

But, I never implemented it, and from where we are moving with warning messages, they don't make much sense now.